File: 4/list-4.py
class Animal:
    def som(self) -> str:
        return "som que animal faz"

class Cachorro(Animal):
    def som(self) -> str:
        return "Au au"

class Gato(Animal):
    def som(self) -> str:
        return "Miau"

def main():
    cachorro = Cachorro()
    gato = Gato()
    print(cachorro.som())
    print(gato.som())

if __name__ == "__main__":
    main()

File: 4/list-4.go
package main

import "fmt"

type Animal interface {
	Som() string
}

type Cachorro struct{}

func (c Cachorro) Som() string {
	return "Au au"
}

type Gato struct{}

func (g Gato) Som() string {
	return "Miau"
}

func fazerSom(a Animal) {
	fmt.Println(a.Som())
}

func main() {
	cachorro := Cachorro{}
	gato := Gato{}

	fazerSom(cachorro)
	fazerSom(gato)
}

